third_party/lua-yaml/lyaml.cc

```diff
@@ -623,6 +618,9 @@ static int dump_node(struct lua_yaml_dumper *dumper)
    bool unused;
    (void) unused;
 
+   yaml_char_t *anchor = get_yaml_anchor(dumper);
+   if (anchor && !*anchor) return 1;
```

Comment:
Shouldn't we do it only for `lua_type(dumper->L, -1) == LUA_TTABLE` for performance matters? It doesn't look worthful to check the `anchortable_index` table for numbers, strings and so on (and we can have quite lot of them to serialize).

Reply:
You're right. Looking up anything but a table in `anchorable_index` is pointless, because `find_references` stores only tables in it:

https://github.com/tarantool/tarantool/blob/d67962828c0eee224935fbe2af2495c7a122c33b/third_party/lua-yaml/lyaml.cc#L781-L784

I updated `get_yaml_anchor` to bail out if the top element isn't a table:

```diff
diff --git a/third_party/lua-yaml/lyaml.cc b/third_party/lua-yaml/lyaml.cc
index 2137ce665f63..bbbfb677063a 100644
--- a/third_party/lua-yaml/lyaml.cc
+++ b/third_party/lua-yaml/lyaml.cc
@@ -511,6 +511,8 @@ usage_error:
 static int dump_node(struct lua_yaml_dumper *dumper);
 
 static yaml_char_t *get_yaml_anchor(struct lua_yaml_dumper *dumper) {
+   if (lua_type(dumper->L, -1) != LUA_TTABLE)
+      return NULL;
    const char *s = "";
    lua_pushvalue(dumper->L, -1);
    lua_rawget(dumper->L, dumper->anchortable_index);
```